bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
+pub struct ObjectMetadata {
+    /// Mode of this object.
+    pub mode: ObjectMode,
+
+    /// Content-Disposition of this object
+    pub content_disposition: Option<String>,
+    /// Content Length of this object
+    pub content_length: Option<u32>,
+    /// Content MD5 of this object.
+    pub content_md5: Option<String>,
+    /// Content Range of this object.
+    pub content_range: Option<Vec<u32>>,
+    /// Content Type of this object.
+    pub content_type: Option<String>,
+    /// ETag of this object.
+    pub etag: Option<String>,
+    /// Last Modified of this object.
+    pub last_modified: i64,
```

Comment:
Does nodejs has native time types?

Reply:
It has. But now `napi-rs` hasn't implemented it yet.

Reply:
Can your link to (or create) an `napi-rs` 's issue of this problem in the comments? So that we know what happened here.

Reply:
I'll do it.

Reply:
Currently, I return an `i64` as a timestamp. (In fact, there is no `timestamp` type in js. It's just a number.)
It's easy to transform to `Date` in js. Users also can set their own time zone.

Reply:
I'm bit concern about this. Return a timestamp will result in the lost of nano seconds. And i16 could exceed `2^53`. So how about return a ISO 8601 string instead, for example: `2014-11-28T21:45:59.324310806+09:00`.

Reply:
Good idea. Will update.

Reply:
Use `DateTime.to_rfc3339()` to implement it. Updated.
```
pub fn to_rfc3339(&self) -> String
Returns an RFC 3339 and ISO 8601 date and time string such as 1996-12-19T16:39:57-08:00.
```

Reply:
I found that `napi-rs` has already implemented the native Date.  My fault.
However, because the native js Date also does not have nanoseconds, we will keep it as a string.